sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/test/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/quickpulse/filteringTest/ValidatorTests.java

```diff
@@ -0,0 +1,218 @@
+package com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filteringTest;
+
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filtering.DerivedMetricProjections;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filtering.Filter;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filtering.KnownRequestColumns;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.filtering.Validator;
+import com.azure.monitor.opentelemetry.autoconfigure.implementation.quickpulse.swagger.models.*;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ValidatorTests {
+
+    @Test
+    void rejectInvalidTelemetryTypes() {
+        List<TelemetryType> telemetryTypes
+            = asList(TelemetryType.EVENT, TelemetryType.PERFORMANCE_COUNTER, TelemetryType.METRIC);
+        Validator validator = new Validator();
+        for (TelemetryType telemetryType : telemetryTypes) {
+            List<FilterConjunctionGroupInfo> filterGroups = createListWithOneFilterConjunctionGroupAndNoFilters();
+            DerivedMetricInfo dmi = createDerivedMetricInfo("random-id", telemetryType.getValue(), AggregationType.SUM,
+                AggregationType.SUM, DerivedMetricProjections.COUNT, filterGroups);
+            assertFalse(validator.isValidDerivedMetricInfo(dmi));
+            DocumentFilterConjunctionGroupInfo docGroup = createDocGroupWithNoFilters(telemetryType);
+            assertFalse(validator.isValidDocConjunctionGroupInfo(docGroup));
+        }
+    }
+
+    @Test
+    void acceptValidTelemetryType() {
+        List<TelemetryType> telemetryTypes
+            = asList(TelemetryType.TRACE, TelemetryType.REQUEST, TelemetryType.DEPENDENCY, TelemetryType.EXCEPTION);
+        Validator validator = new Validator();
+        for (TelemetryType telemetryType : telemetryTypes) {
```

Comment:
You coud use the `ParameterizedTest` JUnit 5 annotation for this test and other ones.